lxc/config_device.go

```diff
@@ -620,12 +621,49 @@ func (c *cmdConfigDeviceSet) Run(cmd *cobra.Command, args []string) error {
 
 		dev, ok := inst.Devices[devname]
 		if !ok {
-			_, ok = inst.ExpandedDevices[devname]
+			expandedDev, ok := inst.ExpandedDevices[devname]
 			if !ok {
 				return fmt.Errorf(i18n.G("Device doesn't exist"))
 			}
 
-			return fmt.Errorf(i18n.G("Device from profile(s) cannot be modified for individual instance. Override device or modify profile instead"))
```

Comment:
We shouldnt be removing this check.
Instead if the root disk is from the profile we'd expect the users to use `lxc config device override <instance> root size.state=`

Also we shouldnt only be doing validation on the client.